services/gitdiff/gitdiff.go

```diff
@@ -485,6 +486,27 @@ func (diff *Diff) LoadComments(ctx context.Context, issue *issues_model.Issue, c
 					sort.SliceStable(line.Comments, func(i, j int) bool {
 						return line.Comments[i].CreatedUnix < line.Comments[j].CreatedUnix
 					})
+
+					// Mark expand buttons that have comments in hidden lines
+					if line.Type == DiffLineSection && line.SectionInfo != nil {
+						var hiddenCommentIDs []int64
+						// Check if there are comments in the hidden range
+						for commentLineNum, comments := range lineCommits {
+							absLineNum := int(commentLineNum)
+							if commentLineNum < 0 {
+								absLineNum = int(-commentLineNum)
+							}
+							if absLineNum > line.SectionInfo.LastRightIdx && absLineNum < line.SectionInfo.RightIdx {
+								// Collect comment IDs
+								for _, comment := range comments {
+									hiddenCommentIDs = append(hiddenCommentIDs, comment.ID)
```

Comment:
```suggestion
									line.HiddenCommentIDs = append(line.HiddenCommentIDs, comment.ID)
```

Reply:
This might actually break functionality, I think its better to just use the temp variable for this logic before adding to line.HiddenCommentIDs.